src/utils/framework.ts

```diff
@@ -399,3 +416,67 @@ export const getMatrix2dToc = (framework: FrameworkFields | undefined): MatrixTo
 
     return toc.filter(isDefined);
 };
+
+export function getScaleWidgetsData(framework: FrameworkFields, entry: Entry) {
+    if (!framework || !framework.widgets) {
+        return [];
+    }
+
+    const { widgets } = framework;
+
+    const scaleWidgets = widgets
+        .filter(w => w.widgetId === 'scaleWidget')
+        .map((w) => {
+            const attributeData = entry.attributes[w.id];
+            const { properties: { data: { scaleUnits } } } = w as WidgetElement<ScaleWidget>;
+            if (isWidgetData(attributeData)) {
+                const value = scaleUnits.find(v => v.key === attributeData.data.value);
+                return value;
+            }
+            return undefined;
+        })
+        .filter(isDefined);
+
+    const scaleWidgetsInsideConditionals = widgets
+        .filter(w => w.widgetId === 'conditionalWidget')
+        .map((conditional) => {
+            const {
+                id,
+                properties: {
+                    data: {
+                        widgets: widgetsInsideConditional = [],
+                    } = {},
+                } = {},
```

Comment:
In typescript, it's better to just use optional chaining instead of setting default values